### PR TITLE
Messages Cluster XML

### PR DESCRIPTION
**Pull request by drempelg**
Adding the XML for the new messages cluster

### CODE REVIEW

**Comment by andy31415** [comment]
@drempelg could you run zap-regen and fix builds?

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    <field name="MessageProtected" mask="0x10"/>
+  </bitmap>
+
+  <struct name="MessageStruct" isFabricScoped="true">
+    <cluster code="0x0097"/>
+    <item fieldId="0" name="MessageID" type="octet_string" length="16" default=""/>
+    <item fieldId="1" name="Priority" type="MessagePriorityEnum"/>
+    <item fieldId="2" name="MessageControl" type="MessageControl" default="0"/>
+    <item fieldId="3" name="StartTime" type="epoch_s" default="0"/>
+    <item fieldId="4" name="Duration" type="int16u" default="0"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:66]: This is also nullable in the spec.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    <item fieldId="5" name="MessageText" type="long_char_string" length="512" default=""/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:67]: Spec has max 256.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    <item fieldId="6" name="Responses" type="MessageResponseOptionStruct" array="true"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:68]: This is optional in the spec.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  </struct>
+
+  <struct name="PresentMessageRequestStruct">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:71]: This does not exist in the spec.

I am stopping here.  Please make this XML match the spec.

File: src/app/zap-templates/zcl/zcl.json [hunk 5]
```diff
@@ -205,6 +206,7 @@
             "UserActiveModeTriggerHint",
             "UserActiveModeTriggerInstruction"
         ],
+        "Messages": ["Messages"],
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl.json:209]: Shouldn't be here.  Why is it here?

File: src/app/zap-templates/zcl/zcl-with-test-extensions.json [hunk 6]
```diff
@@ -207,6 +208,7 @@
             "UserActiveModeTriggerHint",
             "UserActiveModeTriggerInstruction"
         ],
+        "Messages": ["Messages"],
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl-with-test-extensions.json:211]: Shouldn't be here.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
Please note the CI failures.  Some are probably because you do not have `java` installed so the formatting on all the java bits gets messed up.  But some of the other CI failures look real in various ways (failing lints, failing reads on all-clusters-app because of the broken implementation, etc).

In general, the right order of things is:

1. Add XML.
2. Add implementation which is not too broken.
3. Enable implementation in test apps.

File: examples/all-clusters-app/esp32/main/CMakeLists.txt [hunk 7]
```diff
@@ -84,7 +84,8 @@ set(SRC_DIRS_LIST
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/temperature-control-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/time-synchronization-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/dishwasher-alarm-server"
-		      "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/laundry-washer-controls-server"
+            		      "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/laundry-washer-controls-server"
```
[the remark this answers is not in the document]
Comment (lazarkov) [reply examples/all-clusters-app/esp32/main/CMakeLists.txt:87]: Removed.

File: src/app/clusters/messages-server/messages-server.cpp
[the remark this answers is not in the document]
Comment (lazarkov) [reply src/app/clusters/messages-server/messages-server.cpp:1]: Left the dummy code, but removed the implementation from the example apps.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="General"/>
+
+  <bitmap name="Feature" type="bitmap32">
+    <cluster code="0x0097"/>
+    <field name="Confirmation" mask="0x1"/>
```
[the remark this answers is not in the document]
Comment (lazarkov) [reply src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:22]: Updated.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    <field name="ConfirmationResponse" mask="0x2"/>
+    <field name="ConfirmationReply" mask="0x4"/>
+    <field name="ProtectedMessages" mask="0x8"/>
+  </bitmap>
+
+  <enum name="MessagePriorityEnum" type="enum8">
```
[the remark this answers is not in the document]
Comment (lazarkov) [reply src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:28]: Sorted.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+    <item name="Banned" value="0x4"/>
+  </enum>
+
+  <struct name="MessageResponseOptionStruct">
+    <cluster code="0x0097"/>
+    <item fieldId="0" name="MessageResponseID" type="int32u"/>
+    <item fieldId="1" name="Label" type="char_string" length="32" default=""/>
+  </struct>
+
+  <bitmap name="MessageControl" type="bitmap16">
```
[the remark this answers is not in the document]
Comment (lazarkov) [reply src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:51]: Removed the default=""

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    <field name="MessageConfirmed" mask="0x8"/>
+    <field name="MessageProtected" mask="0x10"/>
+  </bitmap>
+
+  <struct name="MessageStruct" isFabricScoped="true">
+    <cluster code="0x0097"/>
+    <item fieldId="0" name="MessageID" type="octet_string" length="16" default=""/>
+    <item fieldId="1" name="Priority" type="MessagePriorityEnum"/>
+    <item fieldId="2" name="MessageControl" type="MessageControl" default="0"/>
+    <item fieldId="3" name="StartTime" type="epoch_s" default="0"/>
```
[the remark this answers is not in the document]
Comment (lazarkov) [reply src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:65]: Updated.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml (hunk 1 above)
Comment (lazarkov) [reply src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:66]: Updated.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml (hunk 2 above)
Comment (lazarkov) [reply src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:67]: Updated.

File: src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml (hunk 3 above)
Comment (lazarkov) [reply src/app/zap-templates/zcl/data-model/chip/messages-cluster.xml:68]: No longer available.

File: src/app/zap-templates/zcl/zcl.json (hunk 5 above)
Comment (lazarkov) [reply src/app/zap-templates/zcl/zcl.json:209]: Removed

**Comment by lazarkov** [comment]
@bzbarsky-apple Continued the development here:
https://github.com/project-chip/connectedhomeip/pull/31656

**Pull request closed without merging**